witness/golang/cmd/witness/impl/witness.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package impl is the implementation of the witness server.
+package impl
```

Comment:
I think this should be in the `internal` directory. `impl` isn't a good name to expose on the public api for the module. I think `cmd/witness/internal/impl/witness.go` would be good?

Reply:
I _think_ that would preclude a do-all-the-things-in-process integration test (like we did in BT/FT demo) - IAFAIU the `internal` visibility rules are that given `/a/b/internal/<stuff>`, <stuff> is only visible to code under `/a/b` (not `/a/z`

That might be ok, just pointing it out (we did `cmd/x/impl` in the FT demo for this reason).

Reply:
Yeah, I'm just using the same layout as in the FT demo.  Happy to change to whatever you two agree is best.